firedrake/mesh.py

```diff
@@ -380,28 +675,21 @@ def add_overlap():
         else:
             raise ValueError("Unknown overlap type %r" % overlap_type)
 
-        dmplex.validate_mesh(plex)
+        dmcommon.validate_mesh(plex)
         plex.setFromOptions()
-        utils._init()
 
         self._topology_dm = plex
-        self.name = name
-        self.comm = dup_comm(plex.comm.tompi4py())
-
-        # A cache of shared function space data on this mesh
-        self._shared_data_cache = defaultdict(dict)
+        self._comm = dup_comm(plex.comm.tompi4py())
```

Comment:
Why did we need to do this? It's because abstractproperties have to be available at class, not instance construction time. Because you're advertising an interface of the class, not the slots of the instance. But you never annotated `comm` as an abstract property, so this dance is unnecessary! The other thing you could do is provide it to the superclass constructor, but I don't really like that because that's an abuse of abstract interfaces. So I would just go with the python duck-typed "beg for forgiveness" option and just set `self.comm` here, and drop the property stuff in this case.

Reply:
If you don't want to do that and do want to provide the property based access, you shouldn't use the internal slot later in this function, but instead once you've set this slot, the public access is available.

Reply:
I set `comm` as a property of the abstract base class because apparently `@abstractproperty` is deprecated - see https://docs.python.org/3/library/abc.html#abc.abstractproperty

Reply:
> If you don't want to do that and do want to provide the property based access, you shouldn't use the internal slot later in this function, but instead once you've set this slot, the public access is available.

I've gone for this option